apollo-ios-codegen/Sources/GraphQLCompiler/CompilationResult.swift

```diff
@@ -214,15 +224,24 @@ public final class CompilationResult: JavaScriptObjectDecodable {
     public var isLocalCacheMutation: Bool {
       directives?.contains { $0.name == Constants.DirectiveNames.LocalCacheMutation } ?? false
     }
+      
+    public var moduleImports: [String] {
+      directives?.compactMap { ImportDirective(directive: $0)?.moduleName } ?? []
+    }
 
     init(_ jsValue: JSValue, bridge: isolated JavaScriptBridge) {
       self.name = jsValue["name"]
       self.type = .fromJSValue(jsValue["typeCondition"], bridge: bridge)
       self.selectionSet = .fromJSValue(jsValue["selectionSet"], bridge: bridge)
-      self.directives = .fromJSValue(jsValue["directives"], bridge: bridge)
       self.referencedFragments = .fromJSValue(jsValue["referencedFragments"], bridge: bridge)
       self.source = jsValue["source"]
       self.filePath = jsValue["filePath"]
+        
+      let directives: [Directive]? = .fromJSValue(jsValue["directives"], bridge: bridge)
```

Comment:
Same as above. We should also try to break this into a static function so that we can re-use the logic instead of repeating it in both places.